content/yaml-publishing/firebase-hosting.md

```diff
@@ -11,6 +11,45 @@ weight: 7
 
 Publishing to Firebase Hosting with Codemagic is a straightforward process as the Firebase CLI is already pre-installed on our virtual machines. Please note that you will have to set it up for your project locally before publishing it to Firebase Hosting. You can find more information in the official [Firebase documentation](https://firebase.google.com/docs/hosting/quickstart).
 
+
+### Using Firebase Google Service Account
+1. To get started you will need a Google service account user with `Cloud Build Service Account`, `Firebase Admin` and `API Keys Admin roles` as shown in the Firebase [docs](https://cloud.google.com/build/docs/deploying-builds/deploy-firebase#required_iam_permissions).
+
+2. You can follow the step-by-step guide to create the service account user [here](../yaml-publishing/firebase-app-distribution/)
+
+3. Configure the `JSON` key received from step 2 as an environment variable name `FIREBASE_SERVICE_ACCOUNT` .
+
+4. Add the path where the contents of the `JSON` key will be copied to as an environment variable name `GOOGLE_APPLICATION_CREDENTIALS` and the value as `$CM_BUILD_DIR/firebase_credentials.json`.
+
+5. Make sure to mark it as `secure` and Enter the variable group name, e.g. **_firebase_credentials_**.
```

Comment:
Perhaps add a link to the page where env var groups are explained

Reply:
i think it will get confusing. Already explained in short how to add here.

Reply:
You are not explaining where the secure checkbox is, what a group is and how to reference it etc. Let's please add a link to that page.

Reply:
✅ done please review
@icarusdust